### ISSUE TITLE
Use computed with getter sends the decorator instead of the value

### ISSUE THREAD

**Issue by GCorbel** [ember-decorators/ember-decorators#200]
I used to have this kind of methods : 

```
import Component from '@ember/component';
import { computed } from 'ember-decorators/object';

export default Component.extend({
  @computed
  get bar() {
    return 'foobar';
  },

  @computed('bar')
  foo(bar) {
    console.log(bar);
    return bar;
  }
});
```

Now, I have the "macro-computed-deprecated" deprecation error so I changed for

```
import Component from '@ember/component';
import { computed } from 'ember-decorators/object';

export default Component.extend({
  @computed
  get bar() {
    return 'foobar';
  },

  @computed('bar')
  get foo() {
    const bar = this.get('bar');
    console.log(bar);
    return bar;
  }
});
```

It works but I have to use getters and I think one of the goals of getters is to avoid that so I tried this : 

```
import Component from '@ember/component';
import { computed } from 'ember-decorators/object';

export default Component.extend({
  @computed
  get bar() {
    return 'foobar';
  },

  @computed('bar')
  get foo() {
    const bar = this.bar;
    console.log(bar);
    return bar;
  }
});
```

In this case, `bar.constructor.name` don't gives `String` but `ComputedProperty`. In other terms, it gives the decorator instead of the value.

I read #146 and I understood `this.bar` should be the normal way. Is there something we can do at the moment?

**Comment by pzuraq** [comment]
So the ability to just use native getters and get the property like in that PR actually required a bit more work behind the scenes, mainly because we don't actually want to expose setters yet (`this.bar = 'foo'`). That's why the [ES5 Getters RFC](https://github.com/emberjs/rfcs/blob/master/text/0281-es5-getters.md) was created and merged recently.

Native getters will be available in Ember 3.1, and we're working on Ember Decorators v2 to bring compatibility for the new paradigm (and TypeScript, and a few other things). You can definitely help if you want! Checkout the [v2 roadmap](https://github.com/ember-decorators/ember-decorators/issues/186) to see what work needs to be done, at the moment we've finished up the preliminary work on `ember-decorators/utils` and launched the beta for that, now we're working on breaking out the individual packages into their own repos and building them on top of the new primitives.

**Comment by GCorbel** [comment]
Thanks for your answer. I would like to contribute but I don't have enough time.

Feel free to close this issue when you want (now or when the v2 will be ready).
